fuseimg/datasets/isic.py

```diff
@@ -0,0 +1,269 @@
+    
+        path = os.path.join(self.data_path, 'ISIC2019/ISIC_2019_Training_Input')
+        print(f"Training Input Path: {os.path.abspath(path)}")
+        if not os.path.exists(path):
+            lgr.info('\nExtract ISIC-2019 training input ... (this may take a few minutes)')
+
+            url = 'https://isic-challenge-data.s3.amazonaws.com/2019/ISIC_2019_Training_Input.zip'
+            wget.download(url, ".")
+            
+            with ZipFile("ISIC_2019_Training_Input.zip", 'r') as zipObj:
+                # Extract all the contents of zip file in current directory
+                zipObj.extractall(path=os.path.join(self.data_path, 'ISIC2019'))
+
+            lgr.info('Extracting ISIC-2019 training input: done')
+
+        path = os.path.join(self.data_path, 'ISIC2019/ISIC_2019_Training_GroundTruth.csv')
+
+        if not os.path.exists(path):
+            lgr.info('\nExtract ISIC-2019 training gt ... (this may take a few minutes)')
+
+            url = 'https://isic-challenge-data.s3.amazonaws.com/2019/ISIC_2019_Training_GroundTruth.csv'
+            wget.download(url, path)
+
+            lgr.info('Extracting ISIC-2019 training gt: done')
+        
+        self._downloaded = True
+
+    def sample_ids(self, size: Optional[int] = None) -> List[str]:
+        """
+        Gets the samples ids in trainset.
+        """
+        images_path = os.path.join(self.data_path, 'ISIC2019/ISIC_2019_Training_Input')
+        
+        samples = [f.split('.')[0] for f in os.listdir(images_path) if f.split('.')[-1] == 'jpg']
+        return samples
+
+    @staticmethod
+    def static_pipeline(data_path: str) -> PipelineDefault:
+
+        static_pipeline = PipelineDefault("static",[
+            # Decoding sample ID
+            (OpISICSampleIDDecode(), dict()),
+            
+            # Load Image
+            (OpLoadImage(data_path), dict(key_in="data.input.img_path", key_out="data.input.img")),
+            
+            # Normalize Image to range [0, 1]
+            (OpNormalizeAgainstSelf(), dict(key="data.input.img")),
+
+            # Cast to numpy array for caching purposes
+            (OpToNumpy(), dict(key="data.input.img")),
+
+            # Read labels into sample_dict. Each class will have a different entry.
+            (OpReadDataframe(data_filename=os.path.join(data_path, '../ISIC_2019_Training_GroundTruth.csv'), key_column='image'), dict()),
+
+            # Squeeze labels into sample_dict['data.label']
+            (OpLambda(func=derive_label), dict())
+        ])
+        return static_pipeline
+
+    @staticmethod
+    def dynamic_pipeline() -> PipelineDefault:
+        """
+        Get suggested dynamic pipeline. including pre-processing that might be modified and augmentation operations. 
+        """
+
+        dynamic_pipeline = PipelineDefault("dynamic", [
+            # Resize images to 3x300x300
+            (OpResizeTo(), dict(key="data.input.img", resize_to=[3, 300, 300])),
```

Comment:
I think (?) that skimage expects channels as the last dim. Please double check that it make sense.

Reply:
I think you are right.  I'll check that.